spec/requests/admin_spec.rb

```diff
@@ -17,23 +17,43 @@
           "provider" => "dfe",
           "info" => {"email" => "test-dfe-sign-in@host.tld"}
         )
-        post admin_dfe_sign_in_path
-        follow_redirect!
       end
 
-      it "renders the admin page and sets a session" do
-        get admin_path
+      context "when the user is authorised to access the service" do
+        before do
+          allow_any_instance_of(DfeSignin::Authorisation).to receive(:has_role?) { true }
+          post admin_dfe_sign_in_path
+          follow_redirect!
+        end
+
+        it "renders the admin page and sets a session" do
+          get admin_path
 
-        expect(response).to be_successful
-        expect(response.body).to include("Admin")
-        expect(session[:login]).to eql({"email" => "test-dfe-sign-in@host.tld"})
+          expect(response).to be_successful
+          expect(response.body).to include("Admin")
+          expect(session[:login]).to eql({"email" => "test-dfe-sign-in@host.tld"})
+        end
+
+        context "and they sign out" do
+          it "unsets the session" do
+            delete admin_sign_out_path
+
+            expect(session[:login]).to be_nil
+          end
+        end
       end
 
-      context "and they sign out" do
-        it "unsets the session" do
-          delete admin_sign_out_path
+      context "when the user is not authorised to access the service" do
+        before do
+          allow_any_instance_of(DfeSignin::Authorisation).to receive(:has_role?) { false }
```

Comment:
As above, let's use `OmniAuth.config.mock_auth[:dfe]=` instead to set a auth hash that doesn't have the expected role instead.